### PR TITLE
[CDAP-14616] Remove all references of Streams in UI

### PR DESCRIPTION
**Pull request by ajainarayanan**
JIRA: https://issues.cask.co/browse/CDAP-14616
Build: https://builds.cask.co/browse/CDAP-UDUT167

**Note:**
- Need to do some exhaustive manual test to check if there are any corner case that is missed.

### CODE REVIEW

File: cdap-ui/app/directives/widget-container/widget-dataset-selector/widget-dataset-selector.js [hunk 1]
```diff
@@ -35,9 +35,7 @@ angular.module(PKG.name + '.commons')
         var resource;
         $scope.list = [];
 
-        if ($scope.datasetType === 'stream') {
-          resource = myStreamApi;
-        } else if ($scope.datasetType === 'dataset') {
+        if ($scope.datasetType === 'dataset') {
```
Comment (elfenheart) [inline cdap-ui/app/directives/widget-container/widget-dataset-selector/widget-dataset-selector.js]: I think at this point we can remove this condition? If it's neither dataset nor stream, it would've thrown an error anyways from line 93

File: cdap-ui/app/directives/widget-container/widget-dataset-selector/widget-dataset-selector.js [hunk 2]
```diff
@@ -140,35 +138,28 @@ angular.module(PKG.name + '.commons')
             return;
           }
 
-          if ($scope.datasetType === 'stream') {
-            params.streamId = $scope.model;
-          } else if ($scope.datasetType === 'dataset') {
+          if ($scope.datasetType === 'dataset') {
```
Comment (elfenheart) [inline cdap-ui/app/directives/widget-container/widget-dataset-selector/widget-dataset-selector.js]: here too

**Comment by elfenheart** [comment]
getting `app/hydrator/services/canvas-factory.js: line 202, col 15, 'isPropertyNotAString' is defined but never used.` on gulp build.

**Review by elfenheart — APPROVED** [review]
👍 LGTM

**Comment by ajainarayanan** [comment]
Latest green build: https://builds.cask.co/browse/CDAP-UDUT167-5
